### DISCUSSION TITLE
Help Request

### DISCUSSION THREAD

**Discussion by lageist** [UNC-Project-Open-AAC/OS-DPI#80] [Q&A] [answered]
[text-based example](https://github.com/UNC-Project-Open-AAC/OS-DPI/blob/f672c79db173298a101a684564959db74872c4c0/Designs%20In%20Progress/Text-based/text_core_key_v10.osdpi)

Need some programming help. I'm attempting the following with the above design. When a selected word has additional forms available in the data, I want them to display in the Forms tab, in the relative location of the selected word. For example, $Display = 'big', Forms tab displays big, bigger, biggest in row 1, columns 9, 10, 11. I've attempted to use  $lastToken and $Display variables and #morph field, but can't figure it out. Any help is appreciated it.

**Comment by gbishop** [comment, marked as answer]
I can meet now if you like. You have several issues that will be easier to
talk through. Then you can write down what they were.

**Reply by lageist** [reply]
Great. I'll send a Zoom.

**Reply by lageist** [reply]
Thanks for your help, @gbishop. I think I've got things working now with the example [here](https://github.com/UNC-Project-Open-AAC/OS-DPI/blob/56b0aed7d01b9c03df8db8724fb8a2ca3ccf8a01/Designs%20In%20Progress/Text-based/text_core_key_v11.osdpi).

Team @sofia-bg, @frrancode, @jbizovi, posted the short Zoom session with @gbishop [here](https://github.com/UNC-Project-Open-AAC/OS-DPI/blob/4a3442e1fb382f987fdbfadc1c5d97a96a39b0dd/Designs%20In%20Progress/Text-based/Gary_help_session_030122.mp4). 

A few things that I was missing: 

1. Referencing forms without having this defined as the grid name. Updated to the grid name 'forms'. It's the name of the grid that matters, not the Tab control, because the event is originating there. 

![Screen Shot 2022-03-01 at 5 53 48 PM](https://user-images.githubusercontent.com/91340176/156262596-4f8b44f7-1ced-4d2c-a8af-e8b0cf7ffdd4.png)

2. Set $lastToken. I was referencing this variable to show the forms based on the last word selected, but I did not have this being set anywhere. Added this to the actions with the add_word rule, so that when a word is pressed on the word grid it is set as the last token (i.e., selected word) so the system remembers. 

![Screen Shot 2022-03-01 at 5 51 43 PM](https://user-images.githubusercontent.com/91340176/156262400-285038aa-b94f-4eb9-8ab3-45cbde474b43.png)

3. Displaying the forms in the same relative location as the selected word. The default is to size the grid to the amount of content data, which makes good sense for lots of use cases. The work around here is to add something to the very last cell. In this case, it's a 7x12 grid, so added * to the cell referencing row 7, column 12.

![Screen Shot 2022-03-01 at 6 00 56 PM](https://user-images.githubusercontent.com/91340176/156263354-7641322a-b1c0-45eb-ac1f-d2845edcb9d1.png)
